lib/wasix/src/lib.rs

```diff
@@ -209,6 +222,33 @@ impl WasiRuntimeError {
     }
 }
 
+/// Wrapper for [`anyhow::Error`] that implements [`std::error::Error`].
+pub struct AnyhowStdError(pub anyhow::Error);
+
+impl std::fmt::Debug for AnyhowStdError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::fmt::Display for AnyhowStdError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::error::Error for AnyhowStdError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        Some(&*self.0)
+    }
+}
+
+impl From<anyhow::Error> for AnyhowStdError {
+    fn from(error: anyhow::Error) -> Self {
+        Self(error)
+    }
+}
```

Comment:
I don't think adding a wrapper around `anyhow::Error` is going to help us much here. The `anyhow` crate is 1.0 so we can rely on it being stable, which means introducing a newtype like this feels like we're adding friction unnecessarily.

There is a `impl From<Error> for Box<dyn std::error::Error + Send + Sync + 'static>` so we should be able to convert from `anyhow::Error` to `SpawnError::Other` directly:

```rust
impl From<anyhow::Error> for SpawnError {
  fn from(e: anyhow::Error) -> SpawnError {
    SpawnError::Other(e.into())
  }
}
```

Reply:
I forget about that `From` impl every damn time...